### ISSUE TITLE
Hyperparameters

### ISSUE THREAD

**Issue by pujacomputes** [AnanyaKumar/transfer_learning#3]
Hi! Thanks for the great code base and cool work. 

I was wondering if the hyper-parameter configurations which were used to report the experiments were available somewhere? If I understand correctly, the configuration files provided are not the final ones since,`"run_adaptation_experiments runs a sweep over these configs and therefore modifies the hyperparameters on configs."`

I'm having a bit of trouble replicating results (especially on Domainnet) and this would really help!

**Comment by AnanyaKumar** [comment]
Thanks for the question! In general, you can run scripts/run_adaptation_experiments.py and comment out https://github.com/AnanyaKumar/transfer_learning/blob/main/scripts/run_adaptation_experiments.py#L26 to get the exact commands run in the sweep.

We only swept over the learning rates. For fine-tuning the best learning rate was 0.001. For LP-FT, the best learning rate was 3e-07 and you need to set --use_net_val_mode=True. For DomainNet, we used a CLIP ResNet-50. The full commands to reproduce the run are below.

The DomainNet fine-tuning command is:

python unlabeled_extrapolation//baseline_train.py --optimizer.args.lr=0.001 --seed=0 --model.classname=models.clip_model.ClipModel --model.args.model_name=RN50 --config=configs//adaptation/domainnet.yaml --root_prefix=/scr/biggest/ --log_dir=logs//full_ft_domainnet_clip_resnet50/optimizer.args.lr-0.001_seed-0_run0 --tmp_par_ckp_dir=/scr/biggest/ue//full_ft_domainnet_clip_resnet50_optimizer.args.lr-0.001_seed-0_run0 --project_name=finetuning --group_name=full_ft_domainnet_clip_resnet50 --run_name=optimizer.args.lr-0.001_seed-0_run0

For LP-FT, we need to first run the linear probing step:

python unlabeled_extrapolation//extract_features.py --model.classname=models.clip_model.ClipModel --model.args.model_name=RN50 --config=configs//adaptation/domainnet.yaml --save_path=logs//linprobe_domainnet_clip_resnet50//features_0 --root_prefix=/scr/biggest/ --use_test_transforms_for_train=False  && python unlabeled_extrapolation//log_reg_sk.py --load_path=logs//linprobe_domainnet_clip_resnet50//features_0 --results_save_path=logs//linprobe_domainnet_clip_resnet50//stats_0.tsv --weights_save_path=logs//linprobe_domainnet_clip_resnet50//weights_0.pkl --num_reg_values=50 --val_metric=test_acc/sketch_val --seed=0

Then run the LP-FT:

python unlabeled_extrapolation//baseline_train.py --optimizer.args.lr=3e-07 --use_net_val_mode=True --linear_probe_checkpoint_path=logs//linprobe_domainnet_clip_resnet50//weights_2.pkl --seed=2 --model.classname=models.clip_model.ClipModel --model.args.model_name=RN50 --config=configs//adaptation/domainnet.yaml --root_prefix=/scr/biggest/ --log_dir=logs//lp_then_ft_valmode_domainnet_clip_resnet50/optimizer.args.lr-3e-07_seed-2_use_net_val_mode-True_run2 --tmp_par_ckp_dir=/scr/biggest/ue//lp_then_ft_valmode_domainnet_clip_resnet50_optimizer.args.lr-3e-07_seed-2_use_net_val_mode-True_run2 --project_name=finetuning --group_name=lp_then_ft_valmode_domainnet_clip_resnet50 --run_name=optimizer.args.lr-3e-07_seed-2_use_net_val_mode-True_run2

**Comment by AnanyaKumar** [comment]
Let me know if you have any other questions! Hoping to clean up the documentation a bit soon

**Comment by pujacomputes** [comment]
Hi, thanks for clarifying! Turns out I was not setting ```--use_net_val_mode=True``` before, which was causing the model to perform significantly worse when I ran FT after LP. I was wondering, did you need to set ```--use_net_val_mode=True``` for other datasets as well, or was this primarily for Domainnet, since the LP step alone is so effective?

**Comment by AnanyaKumar** [comment]
Glad to hear! We tried setting use_net_val_mode to True and False for both
fine tuning and LP-FT (i.e., we also applied it to the baseline to make the
comparison fair). I think use_net_val_mode=True can help with the
distribution shift a bit because batch norm statistics will change for new
datasets, but it’s worth thinking about more! On the other hand, for
vanilla fine-tuning (not so much for LP-FT), keeping batch norm in
inference mode can make training unstable. Averaged across the datasets,
both options performed similarly for fine tuning but use_net_val_mode=True
was a lot less stable, so for vanilla fine tuning we used
use_net_val_mode=False. But for LP-FT, use_net_val_mode=True was better and
still stable, so we used that for all datasets.

Note that for the vision transformer experiments (e.g., imagenet
experiments) this doesn’t matter, because they use group norm, which
behaves the same in train and val mode. In general, I personally prefer
models that avoid batchnorm (and use group norm or layer norm),
particularly for transfer learning.
